Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/Controller.mo

```diff
@@ -0,0 +1,2275 @@
+    final min=fill(0, nChi),
+    final max=fill(1, nChi),
+    final unit=fill("1", nChi))
+    "Chilled water isolation valve position"
+    annotation(Placement(transformation(extent={{-840,-248},{-800,-208}}),
+      iconTransformation(extent={{-140,-90},{-100,-50}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput TChiWatSupResReq
+    "Chilled water supply temperature setpoint reset request"
+    annotation (Placement(transformation(extent={{-840,-320},{-800,-280}}),
+      iconTransformation(extent={{-140,-110},{-100,-70}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput chiPlaReq
+    "Number of chiller plant cooling requests"
+    annotation (Placement(transformation(extent={{-840,-360},{-800,-320}}),
+      iconTransformation(extent={{-140,-130},{-100,-90}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput uConWatPumSpe[nConWatPum](
+    final min=fill(0, nConWatPum),
+    final max=fill(1, nConWatPum),
+    final unit=fill("1", nConWatPum)) "Current condenser water pump speed"
+    annotation(Placement(transformation(extent={{-840,-410},{-800,-370}}),
+        iconTransformation(extent={{-140,-150},{-100,-110}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uConWatPum[nConWatPum]
+    "Condenser water pump status"
+    annotation (Placement(transformation(extent={{-840,-440},{-800,-400}}),
+      iconTransformation(extent={{-140,-170},{-100,-130}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TOut(
+    final unit="K",
+    displayUnit="degC",
+    final quantity="ThermodynamicTemperature")
+    "Outdoor air dry bulb temperature"
+    annotation (Placement(transformation(extent={{-840,-500},{-800,-460}}),
+      iconTransformation(extent={{-140,-190},{-100,-150}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput uChiCooLoa[nChi](
+    final quantity=fill("HeatFlowRate",nChi),
+    final unit=fill("W", nChi)) if have_WSE
+    "Current chiller cooling load"
+    annotation (Placement(transformation(extent={{-840,-560},{-800,-520}}),
+      iconTransformation(extent={{-140,-210},{-100,-170}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput uFanSpe(
+    final quantity="1",
+    final min=0,
+    final max=1) "Tower fan speed"
+    annotation (Placement(transformation(extent={{-840,-608},{-800,-568}}),
+      iconTransformation(extent={{-140,-230},{-100,-190}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TConWatSup(
+    final unit="K",
+    displayUnit="degC",
+    final quantity="ThermodynamicTemperature") if not closeCoupledPlant
+    "Condenser water supply temperature (condenser entering)"
+    annotation(Placement(transformation(extent={{-840,-680},{-800,-640}}),
+      iconTransformation(extent={{-140,-250},{-100,-210}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput uIsoVal[nTowCel](
+    final min=fill(0, nTowCel),
+    final max=fill(1, nTowCel),
+    final unit=fill("1", nTowCel))
+    "Vector of tower cells isolation valve position"
```

Comment:
> Keep the following inlet and outlet isolation valve points if cooling towers have actuated isolation valves. They may have a valve only on the supply with an oversized equalizer used to avoid valves on the outlet, or they may have valves at both inlet and outlet. Delete otherwise.

- So `uIsoVal` should be optional.
- If present, 2 Boolean inputs should be present (as opposed to 1 real) for each tower (cell): valve closed end switch status, and valve open end switch status.

My understanding is that the guideline makes no distinction between the number of cells and the number of towers.

> §3.2.1.4. Open the head pressure control valve of the test chiller and enable a CW pump and the corresponding number of cooling towers for that stage.

But the table in §5.2.12.1 provides a number of Enabled Tower Cells per Plant Stage.

Also, §4. LIST OF POINTS mentions

> CT inlet (and outlet) isolation valve: 2-position line size valve; typ. ***each tower***

And §5.2.12 mentions

> The tower isolation valves can be on the tower inlet only provided the equalizer (or flume gate) is large enough to allow water to flow from the ***enabled cell*** to the disabled cell(s)...

So we should probably make this clear, for instance by using only a number of cooling towers with a comment that one should count one tower for each cell.